src/app/loadout-builder/filter/FilterBuilds.tsx

```diff
@@ -37,6 +42,93 @@ export default function FilterBuilds({
+  const legendaryShardIcon = legendaryShardDef?.displayProperties.icon;
+  const enhancementPrismIcon = enhancementPrismDef?.displayProperties.icon;
+  const ascendantShardIcon = ascendantShardDef?.displayProperties.icon;
+  const ascendantShardName = ascendantShardDef?.displayProperties.name || 'Ascendant Shard';
+
+  const upgradeOptions: Option<UpgradeSpendTier>[] = useMemo(
+    () => [
+      {
+        key: 'explanation',
+        disabled: true,
+        content: (
+          <div className={styles.materialSpendInfo}>
+            {t('LoadoutBuilder.UpgradeTierExplanation')}
+          </div>
+        ),
+      },
+      {
+        key: UpgradeSpendTier.Nothing.toString(),
+        value: UpgradeSpendTier.Nothing,
+        content: (
+          <div className={styles.upgradeOption}>
+            <div className={styles.materialName}>{t('LoadoutBuilder.Nothing')}</div>
+          </div>
+        ),
+      },
+      {
+        key: UpgradeSpendTier.LegendaryShards.toString(),
+        value: UpgradeSpendTier.LegendaryShards,
+        content: (
+          <div className={styles.upgradeOption}>
+            {legendaryShardIcon && <BungieImage src={legendaryShardIcon} />}
+            <div className={styles.materialName}>
+              {legendaryShardDef?.displayProperties.name || 'Legendary Shard'}
+            </div>
+          </div>
+        ),
+      },
+      {
+        key: UpgradeSpendTier.EnhancementPrisms.toString(),
+        value: UpgradeSpendTier.EnhancementPrisms,
+        content: (
+          <div className={styles.upgradeOption}>
+            {enhancementPrismIcon && <BungieImage src={enhancementPrismIcon} />}
+            <div className={styles.materialName}>
+              {enhancementPrismDef?.displayProperties.name || 'Enhancement Prism'}
+            </div>
+          </div>
+        ),
+      },
+      {
+        key: UpgradeSpendTier.AscendantShardsNotExotic.toString(),
+        value: UpgradeSpendTier.AscendantShardsNotExotic,
+        content: (
+          <div className={styles.upgradeOption}>
+            {ascendantShardIcon && <BungieImage src={ascendantShardIcon} />}
+            <div className={styles.materialName}>
+              {t('LoadoutBuilder.NotExotics', {
+                material: ascendantShardName,
+              })}
+            </div>
+          </div>
+        ),
+      },
+      {
+        key: UpgradeSpendTier.AscendantShards.toString(),
+        value: UpgradeSpendTier.AscendantShards,
+        content: (
+          <div className={styles.upgradeOption}>
+            {ascendantShardIcon && <BungieImage src={ascendantShardIcon} />}
+            <div className={styles.materialName}>{ascendantShardName}</div>
+          </div>
+        ),
+      },
+    ],
+    [
+      ascendantShardIcon,
+      ascendantShardName,
+      enhancementPrismDef?.displayProperties.name,
+      enhancementPrismIcon,
+      legendaryShardDef?.displayProperties.name,
+      legendaryShardIcon,
+    ]
```

Comment:
Why not have this just depend on `defs` and do the lookups inside the `useMemo`?

Reply:
Yeah I don't know why I didn't. It's done now.